src/pe/mod.rs

```diff
@@ -56,10 +56,10 @@ pub struct PE<'a> {
     pub is_lib: bool,
     /// Whether the binary is 64-bit (PE32+)
     pub is_64: bool,
-    /// the entry point of the binary
-    pub entry: usize,
-    /// The binary's RVA, or image base - useful for computing virtual addreses
-    pub image_base: usize,
+    /// The entry point RVA of the binary
+    pub entry: u32,
+    /// The binary's VA, or image base - useful for computing virtual addreses
+    pub image_base: u64,
```

Comment:
is this the actual bug? (that we used a `usize` but on 32-bit systems, this can be `u64`?)

If yes, is it really required for `entry` to be fixed to u32? If there's a compelling reason (For consistency perhaps) then it seems fine, but this is a pretty big breaking change, but it may be worth it if it prevents future issues.

Reply:
Yes, this is a serious hidden bug for long.

Fortunately, RVA calculations (i.e., [TLS callbacks map](https://github.com/m4b/goblin/blob/19d88eccfb96a8bce8ff086e9a1bfccd10f645ca/src/pe/tls.rs#L255-L256)) work even it overflows. (i.e., `0x140000000u64` overflows and trimmed into `0x40000000u32`, then subtract the `u32` RVA)

On the other hand, this is significant correctness issue likewise whenever the consumer wants to use it for something like verification.

I guess this has been invisible for this long because no one really cares about 32-bit builds.